packages/config/src/class-config.php

```diff
@@ -0,0 +1,116 @@
+<?php
```

Comment:
I must say I like the entire config structure - it makes everything very well organized. I wonder how it would work when multiple plugins use it though - how do we prevent from them overwriting each other's configurations, which would lead to unexpected results (like, woo not being initialized because another plugin sets it to not be initialized)?

Reply:
That's a very good point, it escaped my mind, thanks for bringing it up.

So, I guess we need to modify this thing in a way where every consumer declares a dependency, but can't really disable anything. So that when a set of plugins consume various packages, we get a superset of everything that has been declared as a dependency.

What do you think?

Reply:
> So, I guess we need to modify this thing in a way where every consumer declares a dependency, but can't really disable anything. So that when a set of plugins consume various packages, we get a superset of everything that has been declared as a dependency.

This sounds good! Each of those would be disabled by default, and the config interface would allow for each plugin to enable it, but not to disable it. We would still have hooks if we want to invasively do something, but not use them unless really necessary. Sounds good!